bin/tx-prover/README.md

```diff
@@ -1,48 +1,60 @@
 # Miden transaction prover
 
-A service for generating Miden transaction proofs on-demand.
+A service for generating Miden transaction proofs on-demand. It is split in two binaries: worker
+and proxy.
+
+The worker is a gRPC service that can receive transaction witnesses and returns the proof. It can
+only handle one request at a time and returns an error if is already in use.
+
+The proxy uses Cloudflare's Pingora crate, which provides features to create a modular proxy. It is
+meant to handle multiple workers with a queue for each one.
 
 ## Installation
 
-Install the prover binary for production using `cargo`:
+Install the prover worker and proxy binaries for production using `cargo`:
 
 ```sh
-cargo install miden-tx-prover --locked
+cargo install miden-tx-prover-worker --locked
+cargo install miden-tx-prover-proxy --locked
 ```
 
 This will install the latest official version of the prover. You can install a specific version using `--version <x.y.z>`:
 
 ```sh
-cargo install miden-tx-prover --locked --version x.y.z
+cargo install miden-tx-prover-worker --locked --version x.y.z
+cargo install miden-tx-prover-proxy --locked --version x.y.z
 ```
 
-You can also use `cargo` to compile the prover service from the source code if for some reason you need a specific git revision. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only. The incantation is a little different as you'll be targetting this repo instead:
+To install both services at once from the source code using specific git revisions you can use `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only:
 
 ```sh
 # Install from a specific branch
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --branch <branch>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --branch <branch>
```

Comment:
I think we can avoid explicitly mentioning how this can be installed from Github (although I think mentioning it like you did above is fine), but I also don't mind if you prefer to keep these